test/fake_device_factory_test.rb

```diff
@@ -21,11 +21,62 @@
 # find current contact information at www.suse.com.
 
 require_relative "spec_helper"
-require "storage"
-require "y2storage/fake_device_factory"
 
 describe Y2Storage::FakeDeviceFactory do
 
+  it "reads yaml of simple dasd and partition setup" do
+
+    environment = Storage::Environment.new(true, Storage::ProbeMode_NONE, Storage::TargetMode_DIRECT)
+    storage = Storage::Storage.new(environment)
+    staging = storage.staging
+
+    # rubocop:disable Style/StringLiterals
+
+    input = ['---',
+             '- dasd:',
+             '    name: "/dev/sda"',
+             '    size: 256 GiB',
+             '    block_size: 4 KiB',
+             '    partition_table: dasd',
```

Comment:
BTW, what does the reader do if we try to format directly a DASD, or to use it as PV? Is not supported by the class, but is detected as a syntax error?